11-node-frameworks/KononovichGalina/index.js

```diff
@@ -0,0 +1,55 @@
+const koa = require('koa');
+const router = require('koa-router');
+const serve = require('koa-static');
+const app = new koa();
+const session = require('koa-generic-session');
+const init = router();
+// const KoaBody = require('koa-body');
+
+// koaBody = convert(KoaBody());
+
+// init.get('/get', getData);
+
+// function getData(ctx, next) {
+//   console.log('nfj');
+//   ctx.body = 'get data';
+//   next();
+// }
+
+// app.use(init.routes());
+
+
+app.use(serve(__dirname + '/public'));
+
+app.use(async(ctx, next) => {
+  ctx.body = 'get data';
+  await next();
+});
+
+app.use(async(ctx, next) => {
+  ctx.body = 'change data';
+  await next();
+});
+
+app.keys = ['Shh, its a secret!'];
+app.use(session(app));
+app.use(async(ctx, next) => {
```

Comment:
keep in mind that counter should work for any request to the server, even posting form. Split counting and getting the "session counter"